### PR TITLE
fix: 394 yaml strings

### LINKED ISSUE #394: String equivalence in yaml
Two different yaml strings should be treated identically in patterns.

For example, this pattern should match all 3 nodes in the yaml file:

```
language yaml

`'ubuntu-latest'`
```

Sample file:
```
steps:
    - runs_on: "ubuntu-latest"
    - runs_on: 'ubuntu-latest'
    - runs_on: ubuntu-latest

```

It looks like the problem is the underlying AST representations are different in the yaml grammar between single-quote and double-quote strings.

### PR DESCRIPTION
**Pull request by Alex-ley-scrub**
/claim #394 
fixes #394 

verified fix with:
```
cargo test --package grit --test apply -- apply_to_yaml_with_multiple_equivalent_strings --exact --show-output 
```

## Greptile Summary

**This is an auto-generated summary**
--
Hi! Looks like you've reached your API usage limit. You can increase it from your account settings page here: [app.greptile.com/settings/usage](https://app.greptile.com/settings/usage)

### CODE REVIEW

File: crates/core/src/marzano_code_snippet.rs [hunk 1]
```diff
@@ -64,11 +64,19 @@ impl Matcher<MarzanoQueryContext> for MarzanoCodeSnippet {
...
+            let kind_id = node.node.kind_id();
+            if *id == kind_id {
+                return true;
+            }
+            // use equivalence classes to match 'ubuntu-latest' and "ubuntu-latest" in yaml
+            // i.e. to match string_scalar, single_quote_scalar, and double_quote_scalar
+            // see https://github.com/getgrit/gritql/issues/394
+            match p {
+                Pattern::AstLeafNode(p) => p.is_equivalent_class(kind_id),
+                _ => false, // TODO: handle other pattern types?
```
Comment (morgante) [inline crates/core/src/marzano_code_snippet.rs:77]: Our style guide requires expanding this so each arm is explicit.

**Review by morgante — APPROVED** [review]
Looks good, once clippy errors are fixed.

File: crates/core/src/marzano_code_snippet.rs (hunk 1 above)
Comment (Alex-ley-scrub) [reply crates/core/src/marzano_code_snippet.rs]: ok, no worries have done: bf9bb755

I don't yet know in depth, what all the different Pattern enum variants mean. From my understanding, it seemed like it was less relevant for the others so I have just returned `false` for all of the other variants - let me know if that doesn't make sense

Comment (morgante) [reply crates/core/src/marzano_code_snippet.rs]: Yeah that's fine for now. We just like to enumerate them so explicit choices are made when adding new ones.

Comment (Alex-ley-scrub) [reply crates/core/src/marzano_code_snippet.rs]: `AstLeafNode` seems to be the only enum variant that has a concept of
`equivalence_class: Option<LeafEquivalenceClass>`

Comment (Alex-ley-scrub) [reply crates/core/src/marzano_code_snippet.rs]: ok, sounds good!

**Review by morgante — APPROVED** [review]

**Comment by Alex-ley-scrub** [comment]
@morgante - very gentle reminder that this one is has not been "rewarded" yet

**Pull request merged**